components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+            String nameParameter = Utils.getQueryParameter(messageContext, PROCESSOR_NAME_PARAM);
+            if (Objects.nonNull(nameParameter)) {
+                populateMessageProcessorData(messageContext, nameParameter);
+            } else {
+                populateMessageProcessorList(messageContext);
+            }
+            axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        } else if (Utils.isDoingPOST(axis2MessageContext)) {
+            try {
+                if (!JsonUtil.hasAJsonPayload(axis2MessageContext)) {
+                    return false;
+                }
+                JsonObject payload = getJsonPayload(messageContext);
+                if (payload.has(PROCESSOR_NAME_PARAM) && payload.has(STATUS_PROPERTY)) {
+                     changeProcessorStatus(messageContext, payload);
+                }
+            } catch (IOException e) {
+                LOG.error("Error when parsing JSON payload", e);
+                Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Error when parsing JSON payload"));
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Create a JSON response with all available message processors
+     *
+     * @param messageContext
+     **/
+    private void populateMessageProcessorList(MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        Map<String, MessageProcessor> processorMap = synapseConfiguration.getMessageProcessors();
+        JSONObject jsonBody = Utils.createJSONList(processorMap.size());
+        processorMap.forEach((key, value) -> addToJSONList(value, jsonBody.getJSONArray(Constants.LIST)));
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+    }
+
+    /**
+     * Create JSON response with information related a defined message processor
+     * @param messageContext synapse message context
+     * @param name name of the message processor
+     * */
+    private void populateMessageProcessorData(MessageContext messageContext, String name) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        MessageProcessor messageProcessor = synapseConfiguration.getMessageProcessors().get(name);
+        if (Objects.nonNull(messageProcessor)) {
+            Utils.setJsonPayLoad(axis2MessageContext, getMessageProcessorAsJson(messageProcessor));
+        } else {
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Message processor does not exist"));
+        }
+    }
+
+    /**
+     * Adds the specified message processor to a JSON list
+     * @param messageProcessor
```

Comment:
Shall we add a description? This is shown as a warning in the IDE.

Reply:
done